src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/service/UserService.kt

```diff
@@ -64,10 +64,16 @@ class UserService(
       .map { u -> toUserDetail(u, username) }
       .orElseThrow(UserNotFoundException("User $username not found"))
 
+  @Transactional(readOnly = true)
   fun findUsersByFirstAndLastNames(firstName: String, lastName: String): List<UserSummary> =
     userPersonDetailRepository.findAllByStaff_FirstNameIgnoreCaseAndStaff_LastNameIgnoreCase(firstName, lastName)
       .map { it.toUserSummary() }
 
+  @Transactional(readOnly = true)
+  fun findAllByEmailAddress(emailAddress: String): List<UserDetail> =
+    userPersonDetailRepository.findByStaff_EmailsEmail(emailAddress)
+      .map { u -> toUserDetail(u, u.username) }
+
```

Comment:
very minor: could be replaced with { toUserDetails(it, it.username) }

And even better not sure why toUserDetails needs the `username` and the UserPersonDetails since `username` is in the object; so the whole thing could be replaced by

{ ::toUserDetails }  (or whatever is the Kotlin syntax for a method reference)

Reply:
definitely - will update in next PR